File: app/src/main/java/ru/vincetti/vimoney/data/adapters/TabsFragmentPagerAdapter.java
package ru.vincetti.vimoney.data.adapters;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import ru.vincetti.vimoney.transaction.TransactionDebtFragment;
import ru.vincetti.vimoney.transaction.TransactionIncomeFragment;
import ru.vincetti.vimoney.transaction.TransactionSpentFragment;
import ru.vincetti.vimoney.transaction.TransactionTransferFragment;

public class TabsFragmentPagerAdapter extends FragmentPagerAdapter {
    private final Bundle fragmentBundle;

    public TabsFragmentPagerAdapter(FragmentManager fm, Bundle data) {
        super(fm);
        fragmentBundle = data;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return "-";
            case 1:
                return "+";
            case 2:
                return "=";
//            case 3:
//                return "|";
            default:
                return null;
        }
    }

    @Override
    public Fragment getItem(int i) {
        switch (i) {
            case 0:
                final TransactionSpentFragment spentFragment = new TransactionSpentFragment();
                spentFragment.setArguments(this.fragmentBundle);
                return spentFragment;
            case 1:
                final TransactionIncomeFragment incomeFragment = new TransactionIncomeFragment();
                incomeFragment.setArguments(this.fragmentBundle);
                return incomeFragment;
            case 2:
                final TransactionTransferFragment transferFragment = new TransactionTransferFragment();
                transferFragment.setArguments(this.fragmentBundle);
                return transferFragment;
//            case 3:
//                return new TransactionDebtFragment();
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return 3;
    }
}

File: app/src/main/java/ru/vincetti/vimoney/data/adapters/TabsFragmentPagerAdapter.kt
package ru.vincetti.vimoney.data.adapters

import android.os.Bundle

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

import ru.vincetti.vimoney.data.models.TransactionModel
import ru.vincetti.vimoney.ui.transaction.income.TransactionIncomeFragment
import ru.vincetti.vimoney.ui.transaction.spent.TransactionSpentFragment
import ru.vincetti.vimoney.ui.transaction.transfer.TransactionTransferFragment

class TabsFragmentPagerAdapter(
        fm: FragmentManager, private val fragmentBundle: Bundle
) : FragmentPagerAdapter(fm) {

    override fun getPageTitle(position: Int): CharSequence {
        return when (position) {
            TransactionModel.TRANSACTION_TYPE_SPENT_TAB -> "-"
            TransactionModel.TRANSACTION_TYPE_INCOME_TAB -> "+"
            TransactionModel.TRANSACTION_TYPE_TRANSFER_TAB -> "="
//            TransactionModel.TRANSACTION_TYPE_DEBT_TAB->"|"
            else -> "Error"
        }
    }

    override fun getItem(position: Int): Fragment {
        return when (position) {
            1 -> {
                val incomeFragment = TransactionIncomeFragment()
                incomeFragment.arguments = this.fragmentBundle
                incomeFragment
            }
            2 -> {
                val transferFragment = TransactionTransferFragment()
                transferFragment.arguments = this.fragmentBundle
                return transferFragment
            }
//            case 3:
//                return new TransactionDebtFragment();
            else -> {
                val spentFragment = TransactionSpentFragment()
                spentFragment.arguments = this.fragmentBundle
                spentFragment
            }
        }
    }

    override fun getCount(): Int {
        return 3
    }
}
